corehq/pillows/case_search.py

```diff
@@ -205,7 +203,7 @@ def build(self):
                 domains = [domain]
             else:
                 # return changes for all enabled domains
-                domains = domains_needing_search_index()
+                domains = _domains_needing_search_index()
```

Comment:
Ongoing work removes the use of transform functions and has put them into the adapter itself. 
So we need to update `_from_dict` function of `corehq.apps.es.case_search.ElasticCaseSearch`

Reply:
just an FYI that Amit and I chatted offline and realized that he had misread the diff and that the transform function was not modified in this process  :)

Reply:
Yep. No blockers from my side.